OtripleS.Web.Api.Tests.Unit/Services/StudentServiceTests/StudentServiceTests.Logic.cs

```diff
@@ -9,6 +9,31 @@ namespace OtripleS.Web.Api.Tests.Unit.Services.StudentServiceTests
 {
     public partial class StudentServiceTests
     {
+        public class RegisterAsync : StudentServiceTests
+        {
+            [Fact]
+            public async Task RegistersNewStudent()
+            {
+                // given
+                Student student = CreateRandomStudent();
+                storageBrokerMock.Setup(broker => broker.AddStudentAsync(It.IsAny<Student>()))
+                    .ReturnsAsync(student);
+
+                // when
+                Student registeredStudent = await studentService.RegisterAsync(student.Id, student.FirstName, student.MiddleName, student.LastName, student.BirthDate, student.Gender);
+
+                // then
+                storageBrokerMock.Verify(broker => broker.AddStudentAsync(It.IsAny<Student>()), Times.Once());
+
+                Assert.Equal(student.Id, registeredStudent.Id);
+                Assert.Equal(student.FirstName, registeredStudent.FirstName);
+                Assert.Equal(student.MiddleName, registeredStudent.MiddleName);
+                Assert.Equal(student.LastName, registeredStudent.LastName);
+                Assert.Equal(student.BirthDate, registeredStudent.BirthDate);
+                Assert.Equal(student.Gender, registeredStudent.Gender);
+            }
```

Comment:
You can just say 
```csharp
actualStudent.Should().BeEquivelantTo(expectedStudent)

```

Please review this work here for reference:

https://github.com/hassanhabib/SchoolEM/blob/master/SchoolEM.Tests/Services/StudentServiceTests/StudentServiceTests.Logic.cs

Reply:
good point, I was wondering whether there is a shorthand for this. Usually use NUnit which doesn't have this, so I usually just compare the serialized json representations. This is nicer.